src/Agent.Worker/StepsRunner.cs

```diff
@@ -309,16 +310,26 @@ private async Task RunStepAsync(IStep step, CancellationToken jobCancellationTok
             step.ExecutionContext.Section(StringUtil.Loc("StepFinishing", step.DisplayName));
             step.ExecutionContext.Complete();
 
+            bool drainQueuesAfterTask = AgentKnobs.DrainQueuesAfterTask.GetValue<BuiltInDefaultKnobSource>(step.ExecutionContext).AsBoolean();
             try
             {
-                // We need to drain the queues after a task just in case if
-                // there are a lot of items since it can cause some UI hangs.
-                await JobServerQueue.DrainQueues();
+                drainQueuesAfterTask = AgentKnobs.DrainQueuesAfterTask.GetValue(step.ExecutionContext).AsBooleanStrict();
             }
-            catch (Exception ex)
+            catch { }
```

Comment:
Swallowing all exceptions with no logging can make issues very difficult to debug e.g. when unexpected exceptions occur.

If a specific exception / message would cause tons of spam in logs and we want to avoid logging; that's okay, but we should still log anything unexpected.

Reply:
This extra logic was reverted